error: package io.micronaut.serde.support.util does not exist import io.micronaut.serde.support.util.SerdeArgumentConstants

### Issue description

[SerdeArgumentContantsError.zip](https://github.com/user-attachments/files/26240976/SerdeArgumentContantsError.zip)

```
./gradlew build

> Task :compileJava FAILED
SLF4J(W): No SLF4J providers were found.
SLF4J(W): Defaulting to no-operation (NOP) logger implementation
SLF4J(W): See https://www.slf4j.org/codes.html#noProviders for further details.
/Users/sdelamo/github/micronaut-projects/micronaut-guides/build/code/micronaut-java-records/micronaut-java-records-gradle-java/build/generated/sources/annotationProcessor/java/main/example/micronaut/SerdeBookForSaleSerializer.java:8: error: package io.micronaut.serde.support.util does not exist
import io.micronaut.serde.support.util.SerdeArgumentConstants;
                                      ^
/Users/sdelamo/github/micronaut-projects/micronaut-guides/build/code/micronaut-java-records/micronaut-java-records-gradle-java/build/generated/sources/annotationProcessor/java/main/example/micronaut/SerdeBookForSaleDeserializer.java:7: error: package io.micronaut.serde.support.util does not exist
import io.micronaut.serde.support.util.SerdeArgumentConstants;
                                      ^
/Users/sdelamo/github/micronaut-projects/micronaut-guides/build/code/micronaut-java-records/micronaut-java-records-gradle-java/build/generated/sources/annotationProcessor/java/main/example/micronaut/SerdeBookCardSerializer.java:8: error: package io.micronaut.serde.support.util does not exist
import io.micronaut.serde.support.util.SerdeArgumentConstants;
                                      ^
/Users/sdelamo/github/micronaut-projects/micronaut-guides/build/code/micronaut-java-records/micronaut-java-records-gradle-java/build/generated/sources/annotationProcessor/java/main/example/micronaut/SerdeBookCardDeserializer.java:7: error: package io.micronaut.serde.support.util does not exist
import io.micronaut.serde.support.util.SerdeArgumentConstants;
                                      ^
4 errors

[Incubating] Problems report is available at: file:///Users/sdelamo/github/micronaut-projects/micronaut-guides/build/code/micronaut-java-records/micronaut-java-records-gradle-java/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileJava'.
> Compilation failed; see the compiler output below.
  /Users/sdelamo/github/micronaut-projects/micronaut-guides/build/code/micronaut-java-records/micronaut-java-records-gradle-java/build/generated/sources/annotationProcessor/java/main/example/micronaut/SerdeBookForSaleSerializer.java:8: error: package io.micronaut.serde.support.util does not exist
  import io.micronaut.serde.support.util.SerdeArgumentConstants;
                                        ^
  /Users/sdelamo/github/micronaut-projects/micronaut-guides/build/code/micronaut-java-records/micronaut-java-records-gradle-java/build/generated/sources/annotationProcessor/java/main/example/micronaut/SerdeBookForSaleDeserializer.java:7: error: package io.micronaut.serde.support.util does not exist
  import io.micronaut.serde.support.util.SerdeArgumentConstants;
                                        ^
  /Users/sdelamo/github/micronaut-projects/micronaut-guides/build/code/micronaut-java-records/micronaut-java-records-gradle-java/build/generated/sources/annotationProcessor/java/main/example/micronaut/SerdeBookCardSerializer.java:8: error: package io.micronaut.serde.support.util does not exist
  import io.micronaut.serde.support.util.SerdeArgumentConstants;
                                        ^
  /Users/sdelamo/github/micronaut-projects/micronaut-guides/build/code/micronaut-java-records/micronaut-java-records-gradle-java/build/generated/sources/annotationProcessor/java/main/example/micronaut/SerdeBookCardDeserializer.java:7: error: package io.micronaut.serde.support.util does not exist
  import io.micronaut.serde.support.util.SerdeArgumentConstants;
                                        ^
  4 errors

* Try:
> Check your code and dependencies to fix the compilation error(s)
> Run with --scan to get full insights from a Build Scan (powered by Develocity).

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.4.1/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 979ms
1 actionable task: 1 executed
````

--- a/serde-api/src/main/java/io/micronaut/serde/util/SerdeArgumentConstants.java
+++ b/serde-api/src/main/java/io/micronaut/serde/util/SerdeArgumentConstants.java
@@ -13,7 +13,7 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package io.micronaut.serde.support.util;
+package io.micronaut.serde.util;
 
 import io.micronaut.core.annotation.Internal;
 import io.micronaut.core.type.Argument;

--- a/serde-jackson/src/test/groovy/io/micronaut/serde/jackson/annotation/SerdeSourceGenEnumSpec.groovy
+++ b/serde-jackson/src/test/groovy/io/micronaut/serde/jackson/annotation/SerdeSourceGenEnumSpec.groovy
@@ -130,6 +130,39 @@ public enum ParityEnum {
         context.close()
     }
 
+    void 'test repeated serde import does not duplicate enum sourcegen files'() {
+        given:
+        def context = buildContext('test.ImportHolder', '''
+package test;
+
+import io.micronaut.serde.annotation.SerdeImport;
+
+@SerdeImport(value = ImportHolder.McpSchema.Role.class)
+@SerdeImport(value = ImportHolder.McpSchema.Role.class)
+public class ImportHolder {
+    public static final class McpSchema {
+        public enum Role {
+            USER,
+            ASSISTANT
+        }
+    }
+}
+''')
+        Class<?> enumType = context.classLoader.loadClass('test.ImportHolder$McpSchema$Role')
+
+        when:
+        def value = Enum.valueOf((Class<Enum>) enumType, 'ASSISTANT')
+        String json = jsonMapper.writeValueAsString(value)
+        def deserialized = jsonMapper.readValue(json, Argument.of(enumType))
+
+        then:
+        json == '"ASSISTANT"'
+        deserialized == value
+
+        cleanup:
+        context.close()
+    }
+
     private static Object deserializeEnum(Deserializer deserializer,
                                           Deserializer.DecoderContext decoderContext,
                                           Argument type,

--- a/serde-processor/src/main/java/io/micronaut/serde/processor/sourcegen/SerdeSourceGenVisitor.java
+++ b/serde-processor/src/main/java/io/micronaut/serde/processor/sourcegen/SerdeSourceGenVisitor.java
@@ -42,6 +42,7 @@ import io.micronaut.sourcegen.model.TypeDef;
 
 import javax.annotation.processing.Generated;
 import javax.lang.model.element.Modifier;
+import java.util.HashSet;
 import java.util.Set;
 
 /**
@@ -54,6 +55,7 @@ public final class SerdeSourceGenVisitor implements TypeElementVisitor<Object, O
     private final RecordSerdeShapeResolver recordSerdeShapeResolver = new RecordSerdeShapeResolver();
     private final BeanSerdeShapeResolver beanSerdeShapeResolver = new BeanSerdeShapeResolver();
     private final EnumSerdeShapeResolver enumSerdeShapeResolver = new EnumSerdeShapeResolver();
+    private final Set<String> writtenGeneratedClassNames = new HashSet<>(32);
 
     @Override
     public Set<String> getSupportedAnnotationNames() {
@@ -89,28 +91,29 @@ public final class SerdeSourceGenVisitor implements TypeElementVisitor<Object, O
     }
 
     private void generateSerializerClass(ClassElement element, SimpleSerdeShapeDecision decision, VisitorContext context) {
+        String generatedSerializerClassName = SerdeSourceGenClassNaming.generatedSerializerClassName(element);
         if (decision.shapeKind() == SimpleSerdeShapeDecision.ShapeKind.RECORD) {
             RecordSerdeShape recordSerdeShape = recordSerdeShapeResolver.resolve(element).orElse(null);
             if (recordSerdeShape != null) {
-                write(context, element, new RecordSerializerSourceGen().generate(element, recordSerdeShape));
+                write(context, element, generatedSerializerClassName, new RecordSerializerSourceGen().generate(element, recordSerdeShape));
                 return;
             }
         }
         if (decision.shapeKind() == SimpleSerdeShapeDecision.ShapeKind.DEFAULT_CONSTRUCTOR_BEAN) {
             BeanSerdeShape beanSerdeShape = beanSerdeShapeResolver.resolve(element).orElse(null);
             if (beanSerdeShape != null) {
-                write(context, element, new BeanSerializerSourceGen().generate(element, beanSerdeShape));
+                write(context, element, generatedSerializerClassName, new BeanSerializerSourceGen().generate(element, beanSerdeShape));
                 return;
             }
         }
         if (decision.shapeKind() == SimpleSerdeShapeDecision.ShapeKind.ENUM) {
             EnumSerdeShape enumSerdeShape = enumSerdeShapeResolver.resolve(element).orElse(null);
             if (enumSerdeShape != null) {
-                write(context, element, new EnumSerializerSourceGen().generate(element, enumSerdeShape));
+                write(context, element, generatedSerializerClassName, new EnumSerializerSourceGen().generate(element, enumSerdeShape));
                 return;
             }
         }
-        write(context, element, ClassDef.builder(SerdeSourceGenClassNaming.generatedSerializerClassName(element))
+        write(context, element, generatedSerializerClassName, ClassDef.builder(generatedSerializerClassName)
             .addModifiers(Modifier.PUBLIC, Modifier.ABSTRACT)
             .addAnnotation(AnnotationDef.builder(Generated.class)
                 .addMember("value", "Micronaut")
@@ -120,28 +123,29 @@ public final class SerdeSourceGenVisitor implements TypeElementVisitor<Object, O
     }
 
     private void generateDeserializerClass(ClassElement element, SimpleSerdeShapeDecision decision, VisitorContext context) {
+        String generatedDeserializerClassName = SerdeSourceGenClassNaming.generatedDeserializerClassName(element);
         if (decision.shapeKind() == SimpleSerdeShapeDecision.ShapeKind.RECORD) {
             RecordSerdeShape recordSerdeShape = recordSerdeShapeResolver.resolve(element).orElse(null);
             if (recordSerdeShape != null) {
-                write(context, element, new RecordDeserializerSourceGen().generate(element, recordSerdeShape));
+                write(context, element, generatedDeserializerClassName, new RecordDeserializerSourceGen().generate(element, recordSerdeShape));
                 return;
             }
         }
         if (decision.shapeKind() == SimpleSerdeShapeDecision.ShapeKind.DEFAULT_CONSTRUCTOR_BEAN) {
             BeanSerdeShape beanSerdeShape = beanSerdeShapeResolver.resolve(element).orElse(null);
             if (beanSerdeShape != null) {
-                write(context, element, new BeanDeserializerSourceGen().generate(element, beanSerdeShape));
+                write(context, element, generatedDeserializerClassName, new BeanDeserializerSourceGen().generate(element, beanSerdeShape));
                 return;
             }
         }
         if (decision.shapeKind() == SimpleSerdeShapeDecision.ShapeKind.ENUM) {
             EnumSerdeShape enumSerdeShape = enumSerdeShapeResolver.resolve(element).orElse(null);
             if (enumSerdeShape != null) {
-                write(context, element, new EnumDeserializerSourceGen().generate(element, enumSerdeShape));
+                write(context, element, generatedDeserializerClassName, new EnumDeserializerSourceGen().generate(element, enumSerdeShape));
                 return;
             }
         }
-        write(context, element, ClassDef.builder(SerdeSourceGenClassNaming.generatedDeserializerClassName(element))
+        write(context, element, generatedDeserializerClassName, ClassDef.builder(generatedDeserializerClassName)
             .addModifiers(Modifier.PUBLIC, Modifier.ABSTRACT)
             .addAnnotation(AnnotationDef.builder(Generated.class)
                 .addMember("value", "Micronaut")
@@ -150,9 +154,13 @@ public final class SerdeSourceGenVisitor implements TypeElementVisitor<Object, O
             .build());
     }
 
-    private void write(VisitorContext context, ClassElement element, ClassDef classDef) {
+    private void write(VisitorContext context, ClassElement element, String generatedClassName, ClassDef classDef) {
+        if (writtenGeneratedClassNames.contains(generatedClassName)) {
+            return;
+        }
         try {
             sourceGenerator.write(classDef, context, element);
+            writtenGeneratedClassNames.add(generatedClassName);
         } catch (Exception e) {
             SourceGenerators.handleFatalException(element, Serdeable.class, e, runtimeException -> {
                 throw runtimeException;

--- a/serde-processor/src/main/java/io/micronaut/serde/processor/sourcegen/beans/BeanSerdeSourceGenUtils.java
+++ b/serde-processor/src/main/java/io/micronaut/serde/processor/sourcegen/beans/BeanSerdeSourceGenUtils.java
@@ -41,7 +41,7 @@ final class BeanSerdeSourceGenUtils {
     private static final String SHORT_TYPE = "short";
 
     private static final TypeDef ARGUMENT_TYPE = TypeDef.of(Argument.class);
-    private static final ClassTypeDef SERDE_ARGUMENT_CONSTANTS = ClassTypeDef.of("io.micronaut.serde.support.util.SerdeArgumentConstants");
+    private static final ClassTypeDef SERDE_ARGUMENT_CONSTANTS = ClassTypeDef.of("io.micronaut.serde.util.SerdeArgumentConstants");
 
     private static final Method ARGUMENT_OF_METHOD = ReflectionUtils.getRequiredMethod(Argument.class, "of", Class.class);
     private static final Method ARGUMENT_OF_WITH_TYPE_PARAMETERS_METHOD = ReflectionUtils.getRequiredMethod(Argument.class, "of", Class.class, Argument[].class);

--- a/serde-processor/src/main/java/io/micronaut/serde/processor/sourcegen/records/RecordSerdeSourceGenUtils.java
+++ b/serde-processor/src/main/java/io/micronaut/serde/processor/sourcegen/records/RecordSerdeSourceGenUtils.java
@@ -41,7 +41,7 @@ final class RecordSerdeSourceGenUtils {
     private static final String SHORT_TYPE = "short";
 
     private static final TypeDef ARGUMENT_TYPE = TypeDef.of(Argument.class);
-    private static final ClassTypeDef SERDE_ARGUMENT_CONSTANTS = ClassTypeDef.of("io.micronaut.serde.support.util.SerdeArgumentConstants");
+    private static final ClassTypeDef SERDE_ARGUMENT_CONSTANTS = ClassTypeDef.of("io.micronaut.serde.util.SerdeArgumentConstants");
 
     private static final Method ARGUMENT_OF_METHOD = ReflectionUtils.getRequiredMethod(Argument.class, "of", Class.class);
     private static final Method ARGUMENT_OF_WITH_TYPE_PARAMETERS_METHOD = ReflectionUtils.getRequiredMethod(Argument.class, "of", Class.class, Argument[].class);

